picasso/src/main/java/com/squareup/picasso/Utils.java

```diff
@@ -283,25 +276,15 @@ static boolean hasPermission(Context context, String permission) {
     return context.checkCallingOrSelfPermission(permission) == PackageManager.PERMISSION_GRANTED;
   }
 
-  static byte[] toByteArray(InputStream input) throws IOException {
-    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-    byte[] buffer = new byte[1024 * 4];
-    int n;
-    while (-1 != (n = input.read(buffer))) {
-      byteArrayOutputStream.write(buffer, 0, n);
-    }
-    return byteArrayOutputStream.toByteArray();
-  }
-
-  static boolean isWebPFile(InputStream stream) throws IOException {
-    byte[] fileHeaderBytes = new byte[WEBP_FILE_HEADER_SIZE];
-    boolean isWebPFile = false;
-    if (stream.read(fileHeaderBytes, 0, WEBP_FILE_HEADER_SIZE) == WEBP_FILE_HEADER_SIZE) {
-      // If a file's header starts with RIFF and end with WEBP, the file is a WebP file
-      isWebPFile = WEBP_FILE_HEADER_RIFF.equals(new String(fileHeaderBytes, 0, 4, "US-ASCII"))
-          && WEBP_FILE_HEADER_WEBP.equals(new String(fileHeaderBytes, 8, 4, "US-ASCII"));
+  static boolean isWebPFile(BufferedSource source) throws IOException {
+    if (source.request(WEBP_FILE_HEADER_SIZE)) {
+      Buffer buffer = source.buffer();
+      if (buffer.rangeEquals(0, WEBP_FILE_HEADER_RIFF)
```

Comment:
you could potentially call `source.rangeEquals()` right? Then you don’t even need to call `request()` above